docs/ReferenceGuides/UnderscoredAttributes.md

```diff
@@ -832,6 +832,106 @@ Fully bypasses access control, allowing access to private declarations
+  formal accesses. In particular, pointers into the storage may be "escaped"
+  outside of scopes where the address is statically guaranteed to be stable
+  and may be used freely for as long as the storage dynamically isn't consumed
+  or moved. It becomes the programmer's responsibility in this case to ensure
+  that reads and writes to the storage do not race across threads, or overlap
+  within the same thread, and that the pointer is not used after the value is
+  moved or consumed.
+- When the value is moved, a bitwise copy of its memory is performed to the
+  new address of the value in its new owner. As currently implemented, raw
+  storage types are not suitable for storing values which are not
+  bitwise-movable, such as nontrivial C++ types or Objective-C weak references.
+
+Using the `@_rawLayout` attribute will suppress the annotated type from
+being implicitly `Sendable`. If the type is safe to access across threads, it
+may be marked `@unchecked Sendable`. This generally means that any
+mutations must be done atomically or with a lock guard, and if the storage
+is ever mutated, then any reads of potentially-mutated state within the storage
+must also be atomic or lock-guarded, because the storage may be accessed
+simultaneously by multiple threads.
+
+A non-Sendable type's memory will be confined to accesses from a single thread
+or task; however, since most mutating operations in Swift still expect
+exclusivity while executing, a programmer must ensure that overlapping
+mutations cannot occur from aliasing, recursion, reentrancy, signal handlers, or
+other potential sources of overlapping access within the same thread.
+
+The parameters to the attribute specify the layout of the type. The following
+forms are currently accepted:
+
+- `@_rawLayout(size: N, alignment: M)` specifies the type's size and alignment
+  in bytes.
+- `@_rawLayout(like: T)` specifies the type's size and alignment should be
+  equal to the type `T`'s.
+- `@_rawLayout(likeArrayOf: T, count: N)` specifies the type's size should be
+  `MemoryLayout<T>.stride * N` and alignment should match `T`'s, like an
+  array of N contiguous elements of `T` in memory.
+
+A notable difference between `@_rawLayout(like: T)` and
+`@_rawLayout(likeArrayOf: T, count: 1)` is that the latter will pad out the
+size of the raw storage to include the full stride of the single element.
+This ensures that the buffer can be safely used with bulk array operations
+despite containing only a single element. `@_rawLayout(like: T)` by contrast
+will exactly match the size and stride of the original type `T`, allowing for
+other values to be stored in the tail padding when the raw layout type appears
+in a larger aggregate.
+
+```
+// struct Weird has size 5, stride 8, alignment 4
+struct Weird {
+    var x: Int32
+    var y: Int8
+}
+
+// struct LikeWeird has size 5, stride 8, alignment 4
+@_rawLayout(like: Weird)
+struct LikeWeird {
+    var x: Int32
```

Comment:
Somewhat weird to see a raw layout type have properties in the example, but I think I understand the intention behind it.

Reply:
Oops, that actually isn't allowed. `@_rawLayout` as I have it implemented precludes the type from having stored properties. Let me fix the example.